icu4c/source/i18n/unicode/coll.h

```diff
@@ -864,7 +864,7 @@ class U_I18N_API Collator : public UObject {
      *         IDs.
      * @stable ICU 2.0
      */
-    virtual UClassID getDynamicClassID(void) const = 0;
+    //virtual UClassID getDynamicClassID(void) const = 0;
```

Comment:
Same question about this being commented out.

Reply:
Notice with the line uncomment, the sublass of Collator are require to implment getDynamicClassID or it will get build/compile time error w/o but if we comment out this line, the superclass version of getDynamicClassID in UObject will kick in and the compiler won't caught such requirement during the compile/build time.

Reply:
Ah, just to make sure I understand -- are you saying that by making `getDynamicClassID` a pure virtual function in the `Collator` class, this will force any subclasses of Collator to implement it? (And by commenting it out here, this now would permit fallback to the superclass `getDynamicClassID` implementation, which isn't what we'd want).

Reply:
sorry, I probably make a mistake while I wrote that comment. It looks like the = 0 line won't do what I though it will.

Reply:
virtual UClassID getDynamicClassID(void) const  override = 0;